src/store/poolSlice.ts

```diff
@@ -100,33 +100,38 @@ export const createPoolSlice: StateCreator<DataStore, [], [], PoolSlice> = (set,
   },
 
   refreshUserData: async (pool_id: string, user: string, latest_ledger_close: number) => {
-    try {
-      const network = get().network;
-      const reserves = get().poolData.get(pool_id)?.reserves;
-      const stellar = get().rpcServer();
+    const network = get().network;
+    const reserves = get().poolData.get(pool_id)?.reserves;
+    const stellar = get().rpcServer();
 
-      if (!reserves) {
-        throw Error('unknown pool');
-      }
-
-      const user_reserve_positions = await UserPositions.load(network, pool_id, user);
-      const userReserveBalances = new Map<string, ReserveBalance>();
-      for (const reserve of reserves) {
-        let userAssetBalance = await getTokenBalance(
-          stellar,
-          network.passphrase,
-          reserve.assetId,
-          Address.fromString(user)
-        );
-        userReserveBalances.set(reserve.assetId, {
-          asset: userAssetBalance,
-          collateral: user_reserve_positions.collateral.get(reserve.config.index) ?? BigInt(0),
-          liability: user_reserve_positions.liabilities.get(reserve.config.index) ?? BigInt(0),
-        });
-      }
+    if (!reserves) {
+      throw Error('unknown pool');
+    }
+    let user_reserve_positions: UserPositions = new UserPositions(new Map(), new Map(), new Map());
+    try {
+      user_reserve_positions = await UserPositions.load(network, pool_id, user);
+    } catch (e) {
+      console.error('Unable to refresh user positions', e);
+    }
```

Comment:
This feels like a bit of a red herring.

The `UserPositions.load` function is expected to fail if the user does not have positions, making an error in console confusing.

I'd recommend keeping the function wide try catch, and instead updating the position call as roughly:

```js
let user_reserve_positions: UserPositions;
try {
    user_reserve_positions = await UserPositions.load(...)
} catch (e) {
    if (e is NotFoundError) {
        user_reserve_positions = new UserPositions(...)
    } else {
        throw e;
    }
}
```

We could also consider pushing this logic down to the sdk, as the load function really shouldn't error for a common scenario

Reply:
This might also point to a bad design decision shoving positions and token balances together.